stubs/regex/regex/regex.pyi

```diff
@@ -0,0 +1,154 @@
+    **kwargs: Any,
+) -> list[AnyStr]: ...
+def splititer(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    string: AnyStr,
+    maxsplit: int = ...,
+    flags: int = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Splitter[AnyStr]: ...
+def findall(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    overlapped: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> list[Any]: ...
+def finditer(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    overlapped: bool = ...,
+    partial: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Scanner[AnyStr]: ...
+def sub(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    repl: AnyStr | Callable[[_regex.Match[AnyStr]], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+def subf(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    format: AnyStr | Callable[[_regex.Match[AnyStr]], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+def subn(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    repl: AnyStr | Callable[[_regex.Match[AnyStr]], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+def subfn(
+    pattern: AnyStr | _regex.Pattern[AnyStr],
+    format: AnyStr | Callable[[_regex.Match[AnyStr]], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> tuple[AnyStr, int]: ...
+def purge() -> None: ...
+@overload
+def cache_all(value: bool = ...) -> None: ...
+@overload
+def cache_all(value: None) -> bool: ...
+def escape(pattern: AnyStr, special_only: bool = ..., literal_spaces: bool = ...) -> AnyStr: ...
+def template(pattern: AnyStr | _regex.Pattern[AnyStr], flags: int = ...) -> _regex.Pattern[AnyStr]: ...
+
+Pattern = _regex.Pattern
+Match = _regex.Match
+Regex = compile
```

Comment:
Missing `Scanner`.

Reply:
So in the source code it doesn't define a type alias for `_regex.Scanner` here.

The `Scanner` that gets exported if you do `from regex import Scanner` is that undocumented `_regex_core.Scanner` class that we discussed removing [here](https://github.com/python/typeshed/pull/6713#discussion_r776119405).

(bit confusing that the package has two Scanner classes)